src/averageAges.js

```diff
@@ -37,7 +46,19 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = people.filter(person => (person.sex === 'f'));
+
+  const womenWithChildren = women.filter(lady =>
+    (people.some(person => lady.name === person.mother) === withChildren));
+
+  const calculateTotalAge = array => (array.reduce((sum, lady) =>
+    (sum + (lady.died - lady.born)), 0));
```

Comment:
move it to separate function so you will be able to reuse it in your functions instead of define it in every function